bittensor/core/async_subtensor.py

```diff
@@ -1235,6 +1235,9 @@ async def get_balances(
         """
         if reuse_block:
             block_hash = self.substrate.last_block_hash
+        elif block_hash is None and block is None:
+            # Neither block nor block_hash provided, default to head
+            block_hash = await self.get_block_hash()
         else:
```

Comment:
```suggestion
         elif block_hash is None:
```
So caller can specify blockhash

Reply:
> So caller can specify blockhash

this logic is covered in `determine_block_hash` method

```py
        # Return the appropriate value.
        if block_hash:
            return block_hash
```